pkg/cloud/aws/dynamodb/client.go

```diff
@@ -67,7 +68,20 @@ func NewClient(ctx context.Context, config cfg.Config, logger log.Logger, name s
 	var err error
 	var awsConfig aws.Config
 
-	if awsConfig, err = gosoAws.DefaultClientConfig(ctx, config, logger, clientCfg.Settings.ClientSettings, clientCfg.LoadOptions...); err != nil {
+	retryOptions := gosoAws.DefaultClientRetryOptions(clientCfg.Settings.ClientSettings)
+	retryOptions = append(retryOptions, gosoAws.RetryWithRetryables([]retry.IsErrorRetryable{
+		&RetryOnTransactionConflict{},
+		&RetryOnConditionalCheckFailed{},
+	}))
+
+	retryer := awsCfg.WithRetryer(func() aws.Retryer {
+		return retry.NewStandard(retryOptions...)
+	})
+
+	configOptions := clientCfg.LoadOptions
+	configOptions = append(configOptions, retryer)
+
+	if awsConfig, err = gosoAws.DefaultClientConfig(ctx, config, logger, clientCfg.Settings.ClientSettings, configOptions...); err != nil {
```

Comment:
I guess the variable should be named `loadOptions` and you could save a line?

```suggestion
	loadOptions := append(clientCfg.LoadOptions, retryer)

	if awsConfig, err = gosoAws.DefaultClientConfig(ctx, config, logger, clientCfg.Settings.ClientSettings, loadOptions...); err != nil {
```